src/classlibnative/bcltype/system.cpp

```diff
@@ -72,23 +72,28 @@ void WINAPI InitializeGetSystemTimeAsFileTime(LPFILETIME lpSystemTimeAsFileTime)
         }
     }
     if (func == NULL)
-#endif
     {
         func = &::GetSystemTimeAsFileTime;
     }
 
-    g_pfnGetSystemTimeAsFileTime = func;
+    InterlockedCompareExchangeT(&g_pfnGetSystemTimeAsFileTime, func, &InitializeGetSystemTimeAsFileTime);
+
     func(lpSystemTimeAsFileTime);
```

Comment:
This should call g_pfnGetSystemTimeAsFileTime or the race condition can still happen.

Reply:
good catch. Thanks!